apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -384,25 +368,19 @@ impl NamedSelection {
 
     pub(crate) fn names(&self) -> Vec<&str> {
         match self {
-            Self::Field(alias, name, _) => {
-                if let Some(alias) = alias {
-                    vec![alias.name.as_str()]
-                } else {
-                    vec![name.as_str()]
-                }
-            }
-            Self::Path { alias, path, .. } => {
+            Self::Field(alias, name, _) => alias
+                .as_ref()
+                .map_or_else(|| vec![name.as_str()], |alias| vec![alias.name.as_str()]),
```

Comment:
I think part of what hurts my brain is that the `None` case is first. The function is called `map_or_else`, but you have to put the `else` before the `map` 😵‍💫. As far as the functions go, I think this conveys intent better:

```suggestion
            Self::Field(alias, name, _) => alias
                .as_ref().map(|alias| vec![alias.name.as_str()])
                .unwrap_or_else(|| vec![name.as_str()]),
```